app/helpers/holdings_helper.rb

```diff
@@ -18,11 +18,18 @@ def holdings(document, method)
   end
 
   def location_name(item)
-    @locations.dig(item[:location].downcase.gsub('_', ''), 'name') || 'Unknown'
+    Location.find_by!(short_code: item[:location].downcase.delete('_').to_sym).name
+  rescue ActiveRecord::RecordNotFound => e
+    Rollbar.error("Error retriving name for Location #{item[:location].downcase.delete('_').to_sym}", e)
+    Location.create(
```

Comment:
I assume `item` here is from Solr/DB right? Not from a user entered param or anything of that nature?

Reply:
The item is one of the item holdings that is returned from Symphony web services. Not from a user entered param. https://github.com/ualbertalib/NEOSDiscovery/blob/c6a5eedb9b3910c428f136144574f55f386b7e2d/app/helpers/holdings_helper.rb#L9-L18